.github/workflows/python_analysis.yml

```diff
@@ -0,0 +1,55 @@
+name: Python analysis
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+    branches:
+      - develop
+      - main
+      - release/**
+      - feature/**
+      - hotfix/**
+  push:
+    branches:
+      - develop
+      - main
+      - release/**
+      - feature/**
+      - hotfix/**
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+env:
+  app_name: 'curve_apps'
+  package_manager: 'conda'
+
+jobs:
+  call-workflow-static-analysis:
+    name: Static analysis
+    uses: MiraGeoscience/CI-tools/.github/workflows/reusable-python-static_analysis.yml@main
+    with:
+      package_manager: 'conda'
+      app_name: ${{ github.env.app_name }}
+      python_vers: '3.10'
+      shell_type: 'bash -l {0}'
```

Comment:
**questison:** is shell_type required?

as the commands within the shared workflow itself depends on the shell type, it does not seem right if the calling workflow has to specify it

Reply:
Yes, it is. Indeed, between poetry and conda, the shell type is not the same.
Within shared workflows, we can always add a step defining the given shell_type, whether it's windows pytest or unix pytest, and whether it's a poetry or conda package.